phoenix-core/src/main/java/org/apache/phoenix/optimize/QueryOptimizer.java

```diff
@@ -395,6 +411,10 @@ private QueryPlan addPlan(PhoenixStatement statement, SelectStatement select, PT
                     }
                 }
             }
+            catch (RowValueConstructorOffsetNotCoercibleException e) {
```

Comment:
Why are we just swallowing the exception here? Should we log something?

Reply:
This is an expected path.  An offset may not be valid for all indexes.